sources/packages/web/src/views/student/StudentProfile.vue

```diff
@@ -5,6 +5,48 @@
     information needs to be changed please visit
     <a href="http://id.gov.bc.ca" target="_blank">id.gov.bc.ca</a>.
   </Message>
+  <span v-if="showApplyPDButton">
+    <v-btn
+      color="primary"
+      @click="applyPDStatus()"
+      v-if="showApplyPDButton"
+      :disabled="disableBtn"
+    >
+      Apply for PD status
+      <span v-if="disableBtn">
+        &nbsp;&nbsp;
+        <ProgressSpinner style="width:30px;height:25px" strokeWidth="10"
+      /></span>
+    </v-btn>
+  </span>
+  <span v-else>
+    <Message
+      severity="warn"
+      :closable="false"
+      v-if="
+        studentAllInfo?.pdSentDate &&
```

Comment:
Just as a recommendation, can we move this logic to a computed property in the setup method to simplify the Vue template?

Reply:
Do we need the null propagation check for these properties?
Is there a scenario where they will not be present? If not, please remove the "?".